### PR TITLE
COMMERCE-7063 An empty string implies user wants to delete the friendlyURL. Delete the friendlyURLEntryLocalization if it exists.

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-7063

@andrea-ale-sbarra ,

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:stop

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java [hunk 1]
```diff
@@ -119,7 +118,7 @@ protected void doProcessAction(
 
 			String urlTitle = urlTitleMap.get(locale);
 
-			if (Validator.isNotNull(urlTitle)) {
+			if (urlTitle != null) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java:121]: Hey @brianikim this change is fine, but you need to add some logic to make sure:

1. the default language translation is not deleted.
2. you don't put an empty string in DB, but you actually delete that record.

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java:121]: Hello @andrea-ale-sbarra , thank you for pointing that out. The updated PR should have all the changes you have requested.

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
Hey @andrea-ale-sbarra ,

The new logic should satisfy both your requirements.

File: modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java [hunk 2]
```diff
@@ -126,6 +127,33 @@ protected void doProcessAction(
 
 				newUrlTitleMap.put(LocaleUtil.toLanguageId(locale), urlTitle);
 			}
+			else if ((urlTitle != null) && urlTitle.equals("")) {
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-asset-categories-web/src/main/java/com/liferay/commerce/product/asset/categories/web/internal/portlet/action/EditAssetCategoryFriendlyURLMVCActionCommand.java:130]: We should use `StringPool.BLANK`
```suggestion
			else if ((urlTitle != null) && urlTitle.equals(StringPool.BLANK)) {
```

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 3]
```diff
@@ -2467,6 +2469,33 @@ private String _getIndexFieldName(String optionKey, String languageId) {
 				newURLTitleMap.put(
 					LocaleUtil.toLanguageId(titleEntry.getKey()), urlTitle);
 			}
+			else if ((urlTitle != null) && urlTitle.equals("")) {
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:2472]: We should use `StringPool.BLANK`
```suggestion
			else if ((urlTitle != null) && urlTitle.equals(StringPool.BLANK)) {
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**